snuba/query/parser/__init__.py

```diff
@@ -305,6 +331,38 @@ def _expand_aliases(query: Query) -> None:
     query.transform(visitor)
 
 
+ARRAYJOIN_FUNCTION_MATCH = FunctionCallMatch(None, StringMatch("arrayJoin"), None)
+
+
+def _validate_arrayjoin(query: Query) -> None:
+    # TODO: Actually validate arrayjoin. For now log how it is used.
+    body_arrayjoin = ""
+    arrayjoin = query.get_arrayjoin_from_ast()
+    if arrayjoin is not None:
+        if isinstance(arrayjoin, Column):
+            body_arrayjoin = arrayjoin.column_name
+
+    array_joins = set()
+    for exp in query.get_all_expressions():
+        match = ARRAYJOIN_FUNCTION_MATCH.match(exp)
+        if match is not None:
+            if isinstance(exp, Column):
+                array_joins.add(exp.column_name)
+            elif isinstance(exp, Literal):
+                array_joins.add(f"{exp.value}")
```

Comment:
Please do not add the value of the literal. That means potentially having a metric per project since new tag values generate new metrics. We want metrics to have a low cardinality.